crates/core/src/db/datastore/system_tables.rs

```diff
@@ -235,231 +248,264 @@ st_fields_enum!(enum StVarFields {
 });
 
 st_fields_enum!(enum StScheduledFields {
-    "table_id", TableId = 0,
-    "reducer_name", ReducerName = 1,
+    "schedule_id", ScheduleId = 0,
+    "table_id", TableId = 1,
+    "reducer_name", ReducerName = 2,
+    "schedule_name", ScheduleName = 3,
 });
 
-/// System Table [ST_TABLE_NAME]
-///
-/// | table_id | table_name  | table_type | table_access |
-/// |----------|-------------|----------- |------------- |
-/// | 4        | "customers" | "user"     | "public"     |
-fn st_table_schema() -> TableSchema {
-    #[allow(deprecated)]
-    TableSchema::from_def(
-        ST_TABLE_ID,
-        RawTableDefV8::new(
-            ST_TABLE_NAME.into(),
-            vec![
-                RawColumnDefV8::sys(StTableFields::TableId.name(), TableId::get_type()),
-                RawColumnDefV8::sys(StTableFields::TableName.name(), AlgebraicType::String),
-                RawColumnDefV8::sys(StTableFields::TableType.name(), AlgebraicType::String),
-                RawColumnDefV8::sys(StTableFields::TablesAccess.name(), AlgebraicType::String),
-            ],
+/// Helper method to check that a system table has the correct fields.
+/// Does not check field types since those aren't included in `StFields` types.
+/// If anything in here is not true, the system is completely broken, so it's fine to assert.
+fn validate_system_table<T: StFields + 'static>(def: &ModuleDef, table_name: &str) {
+    let table = def.table(table_name).expect("missing system table definition");
+    let fields = T::fields();
+    assert_eq!(table.columns.len(), fields.len());
+    for field in T::fields() {
+        let col = table
+            .columns
+            .get(field.col_id().idx())
+            .expect("missing system table field");
+        assert_eq!(&col.name[..], field.name());
+    }
+}
+
+/// See the comment on [`SYSTEM_MODULE_DEF`].
+fn system_module_def() -> ModuleDef {
+    let mut builder = RawModuleDefV9Builder::new();
+
+    // System Table [ST_TABLE_NAME]
+    //
+    // | table_id | table_name  | table_type | table_access |
+    // |----------|-------------|----------- |------------- |
+    // | 4        | "customers" | "user"     | "public"     |
+    let st_table_type = builder.add_type::<StTableRow>();
+    builder
+        .build_table(ST_TABLE_NAME, *st_table_type.as_ref().expect("should be ref"))
+        .with_type(TableType::System)
+        .with_primary_key(StTableFields::TableId)
+        .with_column_sequence(StTableFields::TableId, None)
```

Comment:
A variant `with_primary_key_auto` should exist that also calls the other steps, and is less verbose and error-prone...